issue with npm run dev, "[ERROR] Must use "outdir" when there are multiple input files"

### Describe the bug

Hi, I have followed the get started tutorial of vite. But I could not access the tutorial web page due to the following error

[ERROR] Must use "outdir" when there are multiple input files.

actually I seem to have found out the reason.

if I run the file in 
"/Users/ /Documents/- local files - programming env <Hu>/* - folder, progring projects - test folder/testingArea-hu’s test"
I get error.

but if I instead run it in
"/Users/ /Documents/- local files - programming env <Hu>/ - folder, progring projects - test folder/testingArea-hu’s test"
all things work out.

but why? why does "*" matter.

### Reproduction

https://stackblitz.com/edit/vitejs-vite-mnofsr?terminal=dev

### Steps to reproduce

npm run dev

### System Info

```shell
System:
    OS: macOS 14.2.1
    CPU: (10) arm64 Apple M1 Pro
    Memory: 83.63 MB / 16.00 GB
    Shell: 5.2.2 - /opt/homebrew/bin/bash
  Binaries:
    Node: 21.7.2 - /opt/homebrew/bin/node
    npm: 10.8.1 - /opt/homebrew/bin/npm
    pnpm: 9.9.0 - /opt/homebrew/bin/pnpm
  Browsers:
    Chrome: 128.0.6613.85
    Edge: 128.0.2739.42
    Safari: 17.2.1
```


### Used Package Manager

npm

### Logs

failed to load config from /Users/ /Documents/- local files - programming env <Hu>/* - folder, progring projects - test folder/testingArea-vite, react, django, postgrep/vite-project/vite.config.js
error when starting dev server:
Error: Build failed with 1 error:
error: Must use "outdir" when there are multiple input files
    at failureErrorWithLog (/Users/ /Documents/- local files - programming env <Hu>/* - folder, progring projects - test folder/testingArea-vite, react, django, postgrep/vite-project/node_modules/esbuild/lib/main.js:1472:15)
    at /Users/ /Documents/- local files - programming env <Hu>/* - folder, progring projects - test folder/testingArea-vite, react, django, postgrep/vite-project/node_modules/esbuild/lib/main.js:945:25
    at runOnEndCallbacks (/Users/ /Documents/- local files - programming env <Hu>/* - folder, progring projects - test folder/testingArea-vite, react, django, postgrep/vite-project/node_modules/esbuild/lib/main.js:1315:45)
    at buildResponseToResult (/Users/ /Documents/- local files - programming env <Hu>/* - folder, progring projects - test folder/testingArea-vite, react, django, postgrep/vite-project/node_modules/esbuild/lib/main.js:943:7)
    at /Users/ /Documents/- local files - programming env <Hu>/* - folder, progring projects - test folder/testingArea-vite, react, django, postgrep/vite-project/node_modules/esbuild/lib/main.js:970:16
    at responseCallbacks.<computed> (/Users/ /Documents/- local files - programming env <Hu>/* - folder, progring projects - test folder/testingArea-vite, react, django, postgrep/vite-project/node_modules/esbuild/lib/main.js:622:9)
    at handleIncomingPacket (/Users/ /Documents/- local files - programming env <Hu>/* - folder, progring projects - test folder/testingArea-vite, react, django, postgrep/vite-project/node_modules/esbuild/lib/main.js:677:12)
    at Socket.readFromStdout (/Users/ /Documents/- local files - programming env <Hu>/* - folder, progring projects - test folder/testingArea-vite, react, django, postgrep/vite-project/node_modules/esbuild/lib/main.js:600:7)
    at Socket.emit (node:events:519:28)
    at addChunk (node:internal/streams/readable:559:12)

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitejs/vite/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitejs/vite/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitejs.dev/guide).
- [X] Check that there isn't [already an issue](https://github.com/vitejs/vite/issues) that reports the same bug to avoid creating a duplicate.
- [X] Make sure this is a Vite issue and not a framework-specific issue. For example, if it's a Vue SFC related bug, it should likely be reported to [vuejs/core](https://github.com/vuejs/core) instead.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitejs/vite/discussions) or join our [Discord Chat Server](https://chat.vitejs.dev/).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

--- a/docs/guide/troubleshooting.md
+++ b/docs/guide/troubleshooting.md
@@ -64,6 +64,15 @@ While it may work using [`--experimental-require-module`](https://nodejs.org/doc
 - adding `"type": "module"` to the nearest `package.json`
 - renaming `vite.config.js`/`vite.config.ts` to `vite.config.mjs`/`vite.config.mts`
 
+### `failed to load config from '/path/to/config*/vite.config.js'`
+
+> failed to load config from '/path/to/config\*/vite.config.js'
+> error when starting dev server:
+> Error: Build failed with 1 error:
+> error: Must use "outdir" when there are multiple input files
+
+The error above may occur if the path to your project folder contains `*`, which esbuild treats as a glob. You will need to rename your directory to remove the `*`.
+
 ## Dev Server
 
 ### Requests are stalled forever

